src/pybind/mgr/volumes/fs/async_job.py

```diff
@@ -103,12 +112,28 @@ def __init__(self, volume_client, name_pfx, nr_concurrent_jobs):
         # cv for job cancelation
         self.waiting = False
         self.cancel_cv = threading.Condition(self.lock)
+        self.nr_concurrent_jobs = nr_concurrent_jobs
 
         self.threads = []
         for i in range(nr_concurrent_jobs):
             self.threads.append(JobThread(self, volume_client, name="{0}.{1}".format(name_pfx, i)))
             self.threads[-1].start()
 
+    def reconfigure_max_concurrent_clones(self, name_pfx, nr_concurrent_jobs):
+        """
+        reconfigure number of cloner threads
+        """
+        self.nr_concurrent_jobs = nr_concurrent_jobs
+        # Decrease in concurrency. Notify threads which are waiting for a job to terminate.
+        if len(self.threads) > nr_concurrent_jobs:
```

Comment:
`len(self.threads)` is inspected outside the lock, which basically can report a count that is to be reduced.

IOW, if a decrease in nr_concurrent_jobs is followed by an increase, and a thread is going to remove itself (and hence reduce the thread count) at line 40, but the thread count is inspected here just prior to that being completed, then we will end up with one thread less to process jobs.

Reply:
Reconfigure is serialized at plugin level (src/pybind/mgr/volumes/module.py:398). So this issue should not be happening right?